mms/model_loader.py

```diff
@@ -26,7 +27,20 @@
 MANIFEST_DIR = "manifest_schema"
 MANIFEST_SCHEMA_FILE = 'manifest-schema.json'
 MANIFEST_FILENAME = 'MANIFEST.json'
+LOCK_FILE = '/tmp/tmp_lock_file'
 
+@fasteners.interprocess_locked(LOCK_FILE)
+def _extract_model_archive(model_dir, model_file):
+    if not os.path.isdir(model_dir):
+        os.mkdir(model_dir)
+    try:
+        if '.model' in model_file:
```

Comment:
this check should happen before this method is even called. especially now that getting here requires lock acquisition